larpandoracontent/LArHelpers/LArClusterHelper.cc

```diff
@@ -600,6 +600,59 @@ void LArClusterHelper::GetCoordinateVector(const Cluster *const pCluster, Cartes
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+void LArClusterHelper::GetCaloHitList(const Cluster *const pCluster, CaloHitList &caloHitList)
+{
+    for (const OrderedCaloHitList::value_type &layerEntry : pCluster->GetOrderedCaloHitList())
+        for (const CaloHit *const pCaloHit : *layerEntry.second)
+            caloHitList.push_back(pCaloHit);
+
+    caloHitList.sort(LArClusterHelper::SortHitsByPosition);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void LArClusterHelper::GetCaloHitListInBoundingBox(const pandora::Cluster *const pCluster, const pandora::CartesianVector &lowerBound, 
+        const pandora::CartesianVector &upperBound , pandora::CaloHitList &caloHitList)
+{
+    const bool useX(std::fabs(upperBound.GetX() - lowerBound.GetX()) > std::numeric_limits<float>::epsilon());
+    const bool useY(std::fabs(upperBound.GetY() - lowerBound.GetY()) > std::numeric_limits<float>::epsilon());
+    const bool useZ(std::fabs(upperBound.GetZ() - lowerBound.GetZ()) > std::numeric_limits<float>::epsilon());
+    if (!useX && !useY && !useZ)
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    const float minX(std::min(lowerBound.GetX(), upperBound.GetX()));
+    const float maxX(std::max(lowerBound.GetX(), upperBound.GetX()));
+    const float minY(std::min(lowerBound.GetY(), upperBound.GetY()));
+    const float maxY(std::max(lowerBound.GetY(), upperBound.GetY()));
+    const float minZ(std::min(lowerBound.GetZ(), upperBound.GetZ()));
+    const float maxZ(std::max(lowerBound.GetZ(), upperBound.GetZ()));
+
+    for (const OrderedCaloHitList::value_type &layerEntry : pCluster->GetOrderedCaloHitList())
+    {
+        for (const CaloHit *const pCaloHit : *layerEntry.second)
+        {
+            const CartesianVector hitPosition = pCaloHit->GetPositionVector();
+            if (useX && 
+                (hitPosition.GetX() < minX - std::numeric_limits<float>::epsilon() ||
+                 hitPosition.GetX() > maxX + std::numeric_limits<float>::epsilon()))
+                    continue;
+            else if (useY && 
+                (hitPosition.GetY() < minY - std::numeric_limits<float>::epsilon() ||
+                 hitPosition.GetY() > maxY + std::numeric_limits<float>::epsilon()))
+                    continue;
+            else if (useZ && 
+                (hitPosition.GetZ() < minZ - std::numeric_limits<float>::epsilon() ||
+                 hitPosition.GetZ() > maxZ + std::numeric_limits<float>::epsilon()))
+                    continue;
```

Comment:
I'm not sure I understand the comment @johnmarshall80 
Are you asking for me to change the style because it isn't clear or is what I've done non-standard but clear styling?

Reply:
Hi @absolution1, I was trying to say that this is different to elsewhere in the Pandora codebase, but it looks quite clear (easy to read) and if you like it, it's OK. Our standard style here would probably have each (else) if statement on one line, with parentheses around the resulting code block.

Reply:
OK.  This chunk of if/else if logic is quite big so I'd advocate for clarity here.  I'll leave this one as is.